manager/drivers/secrets.go

```diff
@@ -0,0 +1,51 @@
+package drivers
+
+import (
+	"fmt"
+	"github.com/docker/docker/pkg/plugingetter"
+	"github.com/docker/swarmkit/api"
+)
+
+const (
+	// SecretsProviderAPI is the endpoint for fetching secrets from plugins
+	SecretsProviderAPI = "/SecretProvider.GetSecret"
+
+	// SecretsProviderCapability is the secrets provider plugin capability identification
+	SecretsProviderCapability = "secretprovider"
+)
+
+// SecretDriver provides secrets from different stores
+type SecretDriver struct {
+	plugin plugingetter.CompatPlugin
+}
+
+// NewSecretDriver creates a new driver that provides third party secrets
+func NewSecretDriver(plugin plugingetter.CompatPlugin) *SecretDriver {
+	return &SecretDriver{plugin: plugin}
+}
+
+// Get gets a secret from the secret provider
+func (d *SecretDriver) Get(spec *api.SecretSpec) ([]byte, error) {
```

Comment:
Do we want to provide an `api.SecretSpec` here? Would be great if the driver got information of which service is requesting this secret.